week2/assignment.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strings"
+	"unicode"
+)
+
+type result struct {
+	string
+	int
+}
+
+func fileRead(fi string) map[string]int {
+	data, err := ioutil.ReadFile(fi)
+	if err != nil {
+		log.Fatal(err)
+	}
+	// fmt.Println(string(data))
+	text := string(data)
+
+	fields := strings.FieldsFunc(text, func(c rune) bool {
+		return !unicode.IsLetter(c) && !unicode.IsNumber(c)
+	})
+	words := make(map[string]int)
+	for _, field := range fields {
+		words[strings.ToLower(field)]++
+	}
+	// fmt.Println(words)
+	return words
+}
+
+func folderRead(fo string) []string {
+	var arrDir []string
+	files, err := ioutil.ReadDir(fo)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for _, file := range files {
+		if string(file.Name())[len(string(file.Name()))-3:] == "txt" {
```

Comment:
Some extensions have less/more than 3 characters so that this is not correct, you should split file name by `.` and get the extension from it.